Fix javadoc errors

Javadoc generation is needed for deployment to maven central and it seems JDK 11 is stricter on syntax than previous versions. These will need to be fixed:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:javadoc (default-cli) on project oscar4-core: An error has occurred in Javadoc report generation: 
[ERROR] Exit code: 1 - /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:227: warning: no @param for otherNe
[ERROR] 	public int compareCalculatedConfidenceTo(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:227: warning: no @return
[ERROR] 	public int compareCalculatedConfidenceTo(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:246: warning: no @param for otherNe
[ERROR] 	public int comparePseudoOrCalculatedConfidenceTo(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:246: warning: no @return
[ERROR] 	public int comparePseudoOrCalculatedConfidenceTo(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:267: warning: no @param for otherNe
[ERROR] 	public int compareTypeTo(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:267: warning: no @return
[ERROR] 	public int compareTypeTo(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:277: warning: no @param for otherNe
[ERROR] 	public int compareEnd(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:277: warning: no @return
[ERROR] 	public int compareEnd(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:285: warning: no @param for otherNe
[ERROR] 	public int compareEndToStart(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:285: warning: no @return
[ERROR] 	public int compareEndToStart(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:294: warning: no @param for otherNe
[ERROR] 	public int compareStart(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:294: warning: no @return
[ERROR] 	public int compareStart(NamedEntity otherNe) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:302: warning: no @param for other
[ERROR] 	public int compareStartToEnd(NamedEntity other) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:302: warning: no @return
[ERROR] 	public int compareStartToEnd(NamedEntity other) {
[ERROR] 	           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:342: warning: no @return
[ERROR] 	public double getPseudoConfidence() {
[ERROR] 	              ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:361: warning: no description for @param
[ERROR] 	 * @param deprioritiseOnt
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/NamedEntity.java:407: warning: no description for @param
[ERROR] 	 * @param otherNe
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:35: warning: no @param for instance
[ERROR] 	public IProcessingDocument makeTokenisedDocument(ITokeniser instance,
[ERROR] 	                           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:35: warning: no @param for doc
[ERROR] 	public IProcessingDocument makeTokenisedDocument(ITokeniser instance,
[ERROR] 	                           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:35: warning: no @param for b
[ERROR] 	public IProcessingDocument makeTokenisedDocument(ITokeniser instance,
[ERROR] 	                           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:35: warning: no @param for c
[ERROR] 	public IProcessingDocument makeTokenisedDocument(ITokeniser instance,
[ERROR] 	                           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:35: warning: no @param for d
[ERROR] 	public IProcessingDocument makeTokenisedDocument(ITokeniser instance,
[ERROR] 	                           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:35: warning: no @param for safDoc
[ERROR] 	public IProcessingDocument makeTokenisedDocument(ITokeniser instance,
[ERROR] 	                           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:35: warning: no @return
[ERROR] 	public IProcessingDocument makeTokenisedDocument(ITokeniser instance,
[ERROR] 	                           ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:46: warning: no @return
[ERROR] 	public ProcessingDocument makeTokenisedDocument(ITokeniser tokeniser,
[ERROR] 	                          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:58: warning: no @return
[ERROR] 	public ProcessingDocument makeTokenisedDocument(ITokeniser tokeniser,
[ERROR] 	                          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:80: warning: no @param for tokeniser
[ERROR] 	public ProcessingDocument makeTokenisedDocument(ITokeniser tokeniser,
[ERROR] 	                          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:80: warning: no @param for source
[ERROR] 	public ProcessingDocument makeTokenisedDocument(ITokeniser tokeniser,
[ERROR] 	                          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/ProcessingDocumentFactory.java:80: warning: no @return
[ERROR] 	public ProcessingDocument makeTokenisedDocument(ITokeniser tokeniser,
[ERROR] 	                          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/Token.java:71: warning: no @param for surface
[ERROR]     public void setSurface(String surface) {
[ERROR]                 ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/Token.java:88: warning: no @param for index
[ERROR]     public void setIndex(int index) {
[ERROR]                 ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/Token.java:123: warning: no @param for end
[ERROR]     public void setEnd(int end) {
[ERROR]                 ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/Token.java:140: warning: no @param for bioType
[ERROR] 	public void setBioType(BioType bioType) {
[ERROR] 	            ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/Token.java:157: warning: no @param for tokenSequence
[ERROR]     public void setTokenSequence(TokenSequence tokenSequence) {
[ERROR]                 ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/TokenSequence.java:158: warning: no @param for start
[ERROR]     public String getStringAtOffsets(int start, int end) {
[ERROR]                   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/TokenSequence.java:158: warning: no @param for end
[ERROR]     public String getStringAtOffsets(int start, int end) {
[ERROR]                   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/TokenSequence.java:158: warning: no @return
[ERROR]     public String getStringAtOffsets(int start, int end) {
[ERROR]                   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/TokenSequence.java:257: warning: no @param for index
[ERROR] 	public Token getTokenByStartIndex(int index) {
[ERROR] 	             ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/TokenSequence.java:257: warning: no @return
[ERROR] 	public Token getTokenByStartIndex(int index) {
[ERROR] 	             ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/TokenSequence.java:280: warning: no @param for index
[ERROR] 	public Token getTokenByEndIndex(int index) {
[ERROR] 	             ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/document/TokenSequence.java:280: warning: no @return
[ERROR] 	public Token getTokenByEndIndex(int index) {
[ERROR] 	             ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/exceptions/OscarException.java:16: warning: no description for @param
[ERROR] 	 * @param message
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/exceptions/OscarException.java:25: warning: no description for @param
[ERROR] 	 * @param message
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/exceptions/OscarException.java:26: warning: no description for @param
[ERROR] 	 * @param cause
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/exceptions/DataFormatException.java:17: warning: no description for @param
[ERROR] 	 * @param message
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/exceptions/OscarInitialisationException.java:16: warning: no description for @param
[ERROR] 	 * @param message
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/exceptions/OscarInitialisationException.java:26: warning: no description for @param
[ERROR] 	 * @param message
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/exceptions/OscarInitialisationException.java:27: warning: no description for @param
[ERROR] 	 * @param cause
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/normalize/ITextNormalizer.java:18: warning: no description for @param
[ERROR] 	 * @param character
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/normalize/ITextNormalizer.java:19: warning: no description for @return
[ERROR] 	 * @return
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/normalize/ITextNormalizer.java:28: warning: no description for @return
[ERROR] 	 * @return
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/scixml/TextToSciXML.java:33: error: @param name not found
[ERROR] 	 * @param the {@link XMLStrings} for the output document's schema
[ERROR] 	          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/scixml/TextToSciXML.java:36: warning: no @param for xmlStrings
[ERROR] 	public static SciXMLDocument textToSciXML(String s, XMLStrings xmlStrings) {
[ERROR] 	                             ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/ResourceGetter.java:234: warning: no description for @throws
[ERROR] 	 * @throws IOException
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/StringTools.java:82: warning: no @param for separator
[ERROR]     public static String objectListToString(List<?> l, String separator) {
[ERROR]                          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/StringTools.java:444: error: bad use of '>'
[ERROR]      * A?BC? >>
[ERROR]              ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/StringTools.java:444: error: bad use of '>'
[ERROR]      * A?BC? >>
[ERROR]               ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/StringTools.java:480: warning: no @param for c
[ERROR]     public static String removeCharFromString(char c, String string) {
[ERROR]                          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/StringTools.java:480: warning: no @param for string
[ERROR]     public static String removeCharFromString(char c, String string) {
[ERROR]                          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/StringTools.java:480: warning: no @return
[ERROR]     public static String removeCharFromString(char c, String string) {
[ERROR]                          ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/StringTools.java:527: warning: no description for @param
[ERROR] 	 * @param str
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/StringTools.java:528: warning: no description for @param
[ERROR] 	 * @param suffix
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/tools/StringTools.java:529: warning: no description for @return
[ERROR] 	 * @return
[ERROR] 	   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/types/NamedEntityType.java:64: warning: no @return
[ERROR]     public static synchronized NamedEntityType register(String name, String description) {
[ERROR]                                                ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/types/NamedEntityType.java:80: warning: no @return
[ERROR]     public static synchronized NamedEntityType register(String name, String description, int priority) {
[ERROR]                                                ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/types/NamedEntityType.java:92: warning: no @param for name
[ERROR]     public static NamedEntityType valueOf(String name) {
[ERROR]                                   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/types/NamedEntityType.java:92: warning: no @return
[ERROR]     public static NamedEntityType valueOf(String name) {
[ERROR]                                   ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/types/NamedEntityType.java:136: warning: no @return
[ERROR]     public NamedEntityType getParent() {
[ERROR]                            ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/types/NamedEntityType.java:145: warning: no @param for type
[ERROR]     public boolean isInstance(NamedEntityType type) {
[ERROR]                    ^
[ERROR] /home/travis/build/BlueObelisk/oscar4/oscar4-core/src/main/java/uk/ac/cam/ch/wwmm/oscar/types/NamedEntityType.java:145: warning: no @return
[ERROR]     public boolean isInstance(NamedEntityType type) {
[ERROR]                    ^
[ERROR] 
[ERROR] Command line was: /usr/local/lib/jvm/openjdk11/bin/javadoc @options @packages
```

--- a/pom.xml
+++ b/pom.xml
@@ -348,7 +348,6 @@
                                 <configuration>
                                         <source>8</source>
 					<detectJavaApiLink>false</detectJavaApiLink>
-                                        <doclint>none</doclint>
                                 </configuration>
                         </plugin>
 		</plugins>

